astropy/io/ascii/html.py

```diff
@@ -246,6 +246,15 @@ def end_line(self, lines):
             return None
         return last_index + 1
 
+    def set_fill_values(self, header, cols, fill_values):
```

Comment:
I think you might not need this new method as some of the things being set are probably redundant.  At the place where this gets called, `self.data` is the same as `self` here.  So for instance the `self.data` object already has a copy of header from `__init__`.  So just try moving these statements inline and then remove one-by-one until things break.

The issue is just that all this code is really intricate and intertwined already (sorry), so working hard to minimize complexity is important.